app/api/helpers/payment.py

```diff
@@ -200,6 +200,46 @@ def create_payment(order, return_url, cancel_url):
         else:
             return False, payment.error
 
+    @staticmethod
+    def verify_payment(order, payment_id):
+        try:
+            payment_server = paypalrestsdk.Payment.find(payment_id)
+            if payment_server.state != 'approved':
+                return False, 'Payment has not been approved yet. Status is ' + payment_server.state + '.'
+
+            # Get the most recent transaction
+            transaction = payment_server.transactions[0]
+            amount_server = transaction.amount.total
+            currency_server = transaction.amount.currency
+            sale_state = transaction.related_resources[0].sale.state
+
+            if amount_server != order.amount:
+                return False, 'Payment amount does not match order'
+            elif currency_server != order.event.payment_currency:
+                return False, 'Payment currency does not match order'
+            elif sale_state != 'completed':
+                return False, 'Sale not completed'
+            elif PayPalPaymentsManager.used_payment(payment_id):
+                return False, 'Payment already been verified'
+            else:
+                return True, None
+        except paypalrestsdk.ResourceNotFound:
+            return False, 'Payment Not Found'
+
+    @staticmethod
+    def used_payment(payment_id):
+        """
+        Make sure same payment does not reused
+        """
+        verified_paypal_payment = "paypal_verified_payments.txt"
```

Comment:
Umm, why can't you scan through the model to search for the used payment ID?

Reply:
I'm kinda new to server-side issue so I might need more advices. What do you mean by "scan through the model"? Besides, codecov/patch and codecov/project are failing, what should I do then? @mrsaicharan1

Reply:
@anhanh11001 Yeah, so those checks are ok. Leave them. Travis is important. Scanning through the model as in scanning through the DB table. (SQLAlchemy). Read the SQLalchemy docs and read up on how to query the model & others.

Reply:
In layman terms, would you rather read data from a table in a DB or read the values from a txt file?

Reply:
```
D:\Coding Project\11. Fossasia\open-event-server\venv\lib\site-packages\envparse.py:195: UserWarning: Could not any envfile.
  warnings.warn('Could not any envfile.')
`DATABASE_URL` either not exported or empty
```
I'm getting this when migrating new table, what should I do?

Reply:
This is a warning, not an error